internal/transport/keepalive_test.go

```diff
@@ -310,17 +310,18 @@ func (s) TestKeepaliveClientOpenWithUnresponsiveServer(t *testing.T) {
 // respond to keepalive pings, and makes sure that the client closes the
 // transport even when there is an active stream.
 func (s) TestKeepaliveClientClosesWithActiveStreams(t *testing.T) {
+	pingTime := 500 * time.Millisecond
 	connCh := make(chan net.Conn, 1)
 	copts := ConnectOptions{
 		ChannelzParentID: channelz.NewIdentifierForTesting(channelz.RefSubChannel, time.Now().Unix(), nil),
 		KeepaliveParams: keepalive.ClientParameters{
-			Time:    500 * time.Millisecond,
-			Timeout: 500 * time.Millisecond,
+			Time:    pingTime,
```

Comment:
When we did not have a way to setup a configurable server (like you have added in this diff), we set a "large enough"  time + timeout for KeepaliveParams (`500ms`) to ensure we could atleast create one successful stream before the keepalive timeout kicks and client closes the connection. 

Now since we have more control, i believe we should be able to reduce the timeouts and keep it as low as possible. 

How about setting Time: 10ms and Timeout: 10ms.